### PR TITLE
Making our unlock-js module compatible with our newest version of the smart contracts

### PR DESCRIPTION
**Pull request by julien51**
This was less work/issues than what I expected...
But I think I need to refactor how versions are detected.
1. For Unlock's smart contract, the current version uses opCode which would only yield the address of the proxy... and not the address of the actual implementation...
2. For Locks, we need something custom, and not assume their version based on the version of Unlock being deployed.

These are for the next 2 PRs

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Reviewing by looking at the individual commits will likely be much easier because the bulk of changed files is just renaming of variables...

File: unlock-js/src/v01/getLock.js [hunk 1]
```diff
@@ -48,6 +49,9 @@ export default function(address) {
 
   // Once all lock attributes have been fetched
   return Promise.all(constantPromises).then(() => {
+    // numberOfOwners was previously called outstandingKeys. In order to keep compatibility
+    // we also assign it. This behavior will eventually be deprecated
+    update.outstandingKeys = update.numberOfOwners
```
Comment (hardlydifficult) [inline unlock-js/src/v01/getLock.js]: We renamed outstanding keys to `totalSupply` to follow the ERC721 spec.  `numberOfOwners` may be slightly different

Comment (julien51) [reply unlock-js/src/v01/getLock.js]: Ho, that is a good point. I'll update that! Thanks.

Comment (julien51) [reply unlock-js/src/v01/getLock.js]: Done ;)

**Pull request merged**